src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -98,6 +98,20 @@ public static class PackageDownloader
                 groups.Enqueue(localGroup);
                 groups.Enqueue(otherGroup);
 
+                (bool Prefixmath, HashSet<string> PackageSources) nameSpaceLookupResult = (false, null);
+
+                if (downloadContext?.NameSpaceLookup != null)
+                {
+                    nameSpaceLookupResult = downloadContext.NameSpaceLookup.Find(packageIdentity.Id);
+                    var packageSourcesAtPrefix = nameSpaceLookupResult.Prefixmath ? string.Join(", ", nameSpaceLookupResult.PackageSources) : string.Empty;
+                    File.AppendAllText(@"c:\Test\RestoreLog.txt", "------" + Environment.NewLine);
```

Comment:
You already have a logger here. 

Consider using logger.Debug. 

Some of this logging might stay in the final product too.

Reply:
Ok. Changed to built-in logger.